### PR TITLE
[Functions] (fix): cast fees to uint96 in _calculateCostEstimate

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol [hunk 1]
```diff
@@ -167,7 +167,7 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
 
     uint256 juelsPerGas = _getJuelsPerGas(gasPriceWithOverestimation);
     uint256 estimatedGasReimbursement = juelsPerGas * executionGas;
-    uint96 fees = donFee + adminFee;
+    uint96 fees = uint96(donFee + adminFee);
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:170]: I believe you have to cast each one separately, before adding.

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol]: Yeah, you're right, with that change my little unit test doesn't revert
```
function testUint72Success() public {
    uint72 maxVal = 4_722_366_482_869_645_213_695;
    uint96 bad = uint96(maxVal) + uint96(maxVal);
} 
```

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol [hunk 2]
```diff
@@ -167,7 +167,7 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
 
     uint256 juelsPerGas = _getJuelsPerGas(gasPriceWithOverestimation);
     uint256 estimatedGasReimbursement = juelsPerGas * executionGas;
-    uint96 fees = donFee + adminFee;
+    uint96 fees = uint96(donFee) + uint96(adminFee);
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:170]: I checked that also casting just the first one works correctly (not sure if there's a generally preferred way though).

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol [hunk 3]
```diff
@@ -167,7 +167,7 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
...
 
     return uint96(estimatedGasReimbursement + fees);
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:172]: I was thinking that too. We should have a safecast there, adding

**Review by pinebit — APPROVED** [review]

**Review by RensR — APPROVED** [review]

**Pull request merged**